skimage/filters/edges.py

```diff
@@ -684,11 +684,30 @@ def laplace(image, ksize=3, mask=None):
     skimage.restoration.uft.laplacian().
 
     """
+    def k_size(ksize):
+        if ksize>31:
+            return "Kernel_size Must be less then 31"
+        if ksize%2==0:
+            return "Please use Odd  Kernel_Size"
+        if ksize==3:
+            param=[1,-2,1]
+            return param
+        else:
+            return np.convolve(k_size(ksize-2),[1,-2,1])
+    """
+    D2 Function Genrating Second Derivative 
+    """
+    def d2(input, axis, output, mode, cval ,*extra_arguments):
+        return correlate1d(input, k_size(ksize), axis, output, mode, cval,)
+
+    a = np.zeros((ksize, ksize))
+    a[int(ksize/2), int(ksize/2)] = 1
```

Comment:
This code is 2D only. For nD support it would need to be
```Python
a = np.zeros((ksize,) * image.ndim)
a[(ksize // 2,) * image.ndim] = 1
```

However, I Don't think you need the variables `a` or `x` at all.

I think you can just apply `generic_laplace` directly to image. This will be more efficient than forming a n-dimensional kernels.